src/content/en/updates/2017/03/custom-scrollbar.md

```diff
@@ -0,0 +1,380 @@
+  alt="A piece of CSS code that distorts a div using CSS’
+    perspective attribute.">
+
+The elements inside a perspective container are processed as follows:
+
+* Turn each corner (vertex) of an element into homogenous coordinates
+  `[x,y,z,w]` (relative to the perspective container)
+* Apply all of the element’s transforms as matrices from _right to left_
+* If the perspective element is scrollable, apply a scroll matrix
+* Apply the perspective matrix
+
+The scroll matrix is a translation along the y axis. If we _scroll down_ by
+400px, all elements need to be _moved up_ by 400px. The perspective matrix is a
+matrix that “pulls” points closer to the vanishing point the further back in 3D
+space they are. This achieves both effects of making things appear smaller when
+they are farther back and also makes them “move slower” when being translated.
+So if an element is pushed back, a translation of 400px will cause the element
+to only move 300px on the screen.
+
+If you want to know _all_ the details, you should read the
+[spec](https://drafts.csswg.org/css-transforms-2/#transform-rendering) on CSS’
+transform rendering model, but for the sake of this article I simplified the
+algorithm above.
+
+Our box is inside a perspective container with value p for the `perspective`
+attribute, and let’s assume the container is scrollable and is scrolled down by
+∇ pixels.
+
+<img src="/web/updates/images/2017/03/custom-scrollbar/matrixmath01.svg"
+  alt="Perspective matrix times scroll matrix times element transform matrix
+  equals four by four identity matrix with minus one over p in the fourth row
+  third column time four by four identity matrix with minus nabla in the second
+  row fourth column times element transform matrix.">
+
+The first matrix is the perspective matrix, the second matrix is the scroll
+matrix. To recap: The scroll matrix’ job is to make element _move up_ when we
+are _scrolling down_, hence the negative sign.
+
+For our scrollbar however we want the _opposite_ – we want our element to
+_move down_ when we are scrolling _down_. Here’s where we can use a trick:
+Inverting the _w_ coordinate of the corners of our box. If the _w_ coordinate is
+-1, all translations will take effect in the opposite direction. So how do we do
+that? The CSS engine takes care of converting the corners of our box to
+homogeneous coordinates, and sets w to 1. It’s time for `matrix3d()` to shine!
+
+    .box {
+      transform:
+        matrix3d(
+          1, 0, 0, 0,
+          0, 1, 0, 0,
+          0, 0, 1, 0,
+          0, 0, 0, -1
+        );
+    }
+
+This matrix will do nothing else then negating _w_. So when the CSS engine has
+turned each corner into a vector of the form `[x,y,z,1]`, the matrix will
+convert it into `[x,y,z,-1]`.
+
+<img src="/web/updates/images/2017/03/custom-scrollbar/matrixmath02.svg"
+  alt="Four by four identity matrix with minus one over p in the fourth row
```

Comment:
You've said 'minus one over p' twice now. For someone who studied algebra in English this means -1/p. Is that what you mean?